src/libcore/ops.rs

```diff
@@ -1486,7 +1486,7 @@ shr_assign_impl_all! { u8 u16 u32 u64 usize i8 i16 i32 i64 isize }
 #[lang = "index"]
 #[rustc_on_unimplemented = "the type `{Self}` cannot be indexed by `{Idx}`"]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub trait Index<Idx: ?Sized> {
+pub trait Index<Idx: ?Sized = usize> {
```

Comment:
Can you elaborate on the necessity of this step? I'm slightly wary of changing the global default here, and in principle inference should be able to depend on the impls provided for a particular type.